dynamics/client.py

```diff
@@ -9,11 +9,13 @@
 import os
 from concurrent.futures import ThreadPoolExecutor
 from types import TracebackType
+from typing import Union
 from urllib.parse import quote
 
-from oauthlib.oauth2 import BackendApplicationClient, OAuth2Token
+import httpx
+from authlib.integrations.httpx_client import OAuth2Client
```

Comment:
This pull request is about adding support for the resource keyword. Adding async functionality is a separate issue, and you should make another pull request for that to keep them atomic.

Reply:
I'll split it out